docs/concepts/00-controllers.md

```diff
@@ -0,0 +1,144 @@
+
+The *custodian controller* reconciles periodically. It also reconciles whenever there are changes to
+the `Statefulset` status.
+
+The *custodian controller* is essential to the functioning of etcd-druid, thus the minimum number of
+worker threads is 1, (default being 3).
+
+## compaction controller
+
+The *compaction controller* deploys the snapshot compaction job whenever required.
+The controller watches the number of delta events occurences and triggers a snapshot compaction when the number of
+delta events crosses the set threshold (1M events by default, which is configurable).
+
+The controller watches for changes in "snapshot" `Leases` associated with `Etcd` resources.
+It checks the full and delta snapshot `Leases` and calculates the difference in events between the latest
+delta snapshot and the previous full snapshot, and initiates the compaction job if the event threshold is crossed.
+
+The number of worker threads for the *compaction controller* needs to be greater than or equal to 0 (default 3).
+This is unlike other controllers who need atleast one worker thread for the proper functioning of etcd-druid.
+This is because snapshot compaction is not a core functionality for the etcd clusters to be deployed.
+
+## etcdcopybackupstask controller
+
+The *etcdcopybackupstask controller* is responsible for deploying the snapshot copy job.
+
+Whenever `EtcdCopyBackupsTask` resources are deployed or updated, the *etcdcopybackupstask controller* picks up these
+events from the event queue.
+The `EtcdCopyBackupsTask` uses the etcd-backup-restore image, and deploys it as a job. The argument used for the job
+is the `copy` command of `etcdbrctl`, and has the source and target storage providers as arguments when creating
+the job resource instance.
+
+The number of worker threads for the *etcdcopybackupstask controller* needs to be greater than or equal to 0 (default
+being 3). This is unlike other controllers who need atleast one worker thread for the proper functioning of etcd-druid.
+This is because etcdcopybackupstask is not a core functionality for the etcd clusters to be deployed.
+
+## secret controller
+
+The *secret controller*'s primary responsibility is to add a finalizer on `Secret`s referenced by the `Etcd` resource.
+The *secret controller* is registered for `Secret`s, and the controller keeps a watch on the `Etcd` CR.
+This finalizer is added to ensure that `Secret`s which are referenced by the `Etcd` CR aren't deleted while still
+being used by the `Etcd` resource.
+
+Events which involve the `Etcd` CR which refer to a `Secret` are enqueued into the request queue, which the reconciler
+acts on.  
+The controller reconciles whenever necessary events are queued in, and thus is triggered automatically.
+
+The number of worker threads for the secret controller must be atleast 1 (default being 10), since infrastructure
+access to store snapshots is essential to the proper functioning of etcd-druid.
```

Comment:
```suggestion
The number of worker threads for the secret controller must be atleast 1 (default being 10) for this core controller, since the referenced TLS and infrastructure access secrets are essential to the proper functioning of the etcd cluster.
```